source/module_hamilt_lcao/module_gint/gint_rho_cpu_interface.cpp

```diff
@@ -53,31 +122,136 @@ void Gint::cpu_tau_interface(Gint_inout* inout) {
+                    *this->gridt,this->bxyz, grid_index, na_grid, LD_pool,
+                    block_iw.data(), block_size.data(),
+                    block_index.data(), cal_flag.get_ptr_2D(),
+                    dpsir_ylm_x.get_ptr_2D(),
+                    dpsix_DM.get_ptr_2D(),
+                    this->DMRGint[is], 1);
+                Gint_Tools::mult_psi_DM_new(
+                    *this->gridt, this->bxyz, grid_index, na_grid, LD_pool,
+                    block_iw.data(), block_size.data(),
+                    block_index.data(), cal_flag.get_ptr_2D(),
+                    dpsir_ylm_y.get_ptr_2D(),
+                    dpsiy_DM.get_ptr_2D(),
+                    this->DMRGint[is], 1);	
+                Gint_Tools::mult_psi_DM_new(
+                    *this->gridt, this->bxyz, grid_index, na_grid, LD_pool,
+                    block_iw.data(), block_size.data(),
+                    block_index.data(), cal_flag.get_ptr_2D(),
+                    dpsir_ylm_z.get_ptr_2D(),
+                    dpsiz_DM.get_ptr_2D(),
+                    this->DMRGint[is], 1);
+            }
+            else
+            {
+                Gint_Tools::mult_psi_DMR(
+                    *this->gridt, this->bxyz,
+                    LD_pool,
+                    grid_index, na_grid,
+                    block_index.data(), block_size.data(),
+                    cal_flag.get_ptr_2D(), 
+                    dpsir_ylm_x.get_ptr_2D(),
+                    dpsix_DM.get_ptr_2D(),
+                    this->DMRGint[is],
+                    1);
+                Gint_Tools::mult_psi_DMR(
+                    *this->gridt, this->bxyz,
+                    LD_pool,
+                    grid_index, na_grid,
+                    block_index.data(), block_size.data(),
+                    cal_flag.get_ptr_2D(),
+                    dpsir_ylm_y.get_ptr_2D(),
+                    dpsiy_DM.get_ptr_2D(),
+                    this->DMRGint[is],
+                    1);
+                Gint_Tools::mult_psi_DMR(
+                    *this->gridt, this->bxyz,
+                    LD_pool,
+                    grid_index, na_grid,
+                    block_index.data(), block_size.data(),
+                    cal_flag.get_ptr_2D(), 
+                    dpsir_ylm_z.get_ptr_2D(),
+                    dpsiz_DM.get_ptr_2D(),
+                    this->DMRGint[is],
+                    1);
+            }
+
+        //do sum_i,mu g_i,mu(r) * d/dx_i psi_mu(r) to get kinetic energy density on grid
+            if(inout->job==Gint_Tools::job_type::tau)
+            {
+                this->cal_meshball_tau(
+                    na_grid, block_index.data(),
+                    vindex.data(),
+                    dpsir_ylm_x.get_ptr_2D(), dpsir_ylm_y.get_ptr_2D(), dpsir_ylm_z.get_ptr_2D(),
+                    dpsix_DM.get_ptr_2D(), dpsiy_DM.get_ptr_2D(), dpsiz_DM.get_ptr_2D(),
+                    inout->rho[is]);
+            }
+        }
     }
+#ifdef _OPENMP
+}
+#endif
```

Comment:
This piece of code looks odd, I believe there is a better way to refactor it.

Reply:
About which part? OpenMP or Gint's code?

Reply:
The issue of removing the OpenMP judgment condition has been resolved.